### PR TITLE
doc: update snap-clone with rbd snapshot

### PR DESCRIPTION
**Pull request by yati1998**
RBD Snapshot doc was the part of the README.md file. Hence,
renamed the cephfs-snap-clone.md file to snap-clone.md file
and moved the rbd snapshot document there.

Signed-off-by: yati padia <ypadia@redhat.com>

Fixes: #1533

### CODE REVIEW

File: scripts/minikube.sh
Comment (Yuggupta27) [inline scripts/minikube.sh:79]: Was this change intended?

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]

File: scripts/minikube.sh
Comment (yati1998) [reply scripts/minikube.sh:79]: No No, I did for my local change. Also, havent added this for commit. Will update it. Thanks for informing

Comment (Yuggupta27) [reply scripts/minikube.sh:79]: No issues :)

File: docs/snap-clone.md
Comment (Yuggupta27) [inline docs/snap-clone.md:110]: These links don't seem to work for me

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]

File: docs/snap-clone.md
Comment (yati1998) [reply docs/snap-clone.md:112]: Okay, let me check. I have mentioned as it was mentioned in README.md

File: docs/snap-clone.md
Comment (Yuggupta27) [inline docs/snap-clone.md:112]: To make it look cleaner, you can use labels and list the label+url at the bottom of the file

**Review by Yuggupta27 — CHANGES_REQUESTED** [review]

**Comment by Yuggupta27** [comment]
You can run `make static-check` locally to resolve errors and make travis-ci happy :)

**Comment by yati1998** [comment]
> You can run `make static-check` or ` make containerized-test` locally to resolve errors and make `travis-ci` happy :)

yes I have done that. checking for some error again.

**Comment by nixpanic** [comment]
The CI job fails with the following error:
```
mdl --style scripts/mdl-style.rb ./docs/snap-clone.md 
./docs/snap-clone.md:17: MD012 Multiple consecutive blank lines
./docs/snap-clone.md:206: MD013 Line length
./docs/snap-clone.md:208: MD013 Line length
./docs/snap-clone.md:209: MD013 Line length
./docs/snap-clone.md:18: MD025 Multiple top level headers in the same document
./docs/snap-clone.md:105: MD025 Multiple top level headers in the same document
./docs/snap-clone.md:127: MD026 Trailing punctuation in header
./docs/snap-clone.md:135: MD026 Trailing punctuation in header
./docs/snap-clone.md:141: MD026 Trailing punctuation in header
./docs/snap-clone.md:149: MD026 Trailing punctuation in header
A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
mdl --style scripts/mdl-style.rb ./docs/deploy-rbd.md 
Makefile:96: recipe for target 'lint-extras' failed
make: *** [lint-extras] Error 123
The command "make lint-extras" exited with 2.
```

You can run `make containerized-test TARGET=lint-extras` locally to see if changes to your branch fix these.

**Comment by yati1998** [comment]
> The CI job fails with the following error:
> 
> ```
> mdl --style scripts/mdl-style.rb ./docs/snap-clone.md 
> ./docs/snap-clone.md:17: MD012 Multiple consecutive blank lines
> ./docs/snap-clone.md:206: MD013 Line length
> ./docs/snap-clone.md:208: MD013 Line length
> ./docs/snap-clone.md:209: MD013 Line length
> ./docs/snap-clone.md:18: MD025 Multiple top level headers in the same document
> ./docs/snap-clone.md:105: MD025 Multiple top level headers in the same document
> ./docs/snap-clone.md:127: MD026 Trailing punctuation in header
> ./docs/snap-clone.md:135: MD026 Trailing punctuation in header
> ./docs/snap-clone.md:141: MD026 Trailing punctuation in header
> ./docs/snap-clone.md:149: MD026 Trailing punctuation in header
> A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
> mdl --style scripts/mdl-style.rb ./docs/deploy-rbd.md 
> Makefile:96: recipe for target 'lint-extras' failed
> make: *** [lint-extras] Error 123
> The command "make lint-extras" exited with 2.
> ```
> 
> You can run `make containerized-test TARGET=lint-extras` locally to see if changes to your branch fix these.

Thanks @nixpanic will check and update it.

File: docs/snap-clone.md
Comment (Madhu-1) [inline docs/snap-clone.md:187]:
```suggestion
To restore the snapshot to a new PVC, create
[pvc-restore.yaml](../examples/rbd/pvc-restore.yaml)
and a testing pod [pod-restore.yaml](../examples/rbd/pod-restore.yaml)
```

**Review by Madhu-1 — DISMISSED** [review]

**Comment by nixpanic** [comment]
@mergifyio rebase

File: docs/snap-clone.md
Comment (nixpanic) [inline docs/snap-clone.md:53]: Is this still needed? You improved the name of the title, so I think this can be removed now.

**Review by nixpanic — COMMENTED** [review]
As mentioned on an other PR, I would really like to see your full name in the `Signed-off-by` instead of your GitHub username.

**Comment by yati1998** [comment]
> As mentioned on an other PR, I would really like to see your full name in the `Signed-off-by` instead of your GitHub username.

yes, I have done that already, in the commit message.

**Comment by nixpanic** [comment]
> > As mentioned on an other PR, I would really like to see your full name in the `Signed-off-by` instead of your GitHub username.
> 
> yes, I have done that already, in the commit message.

Hmm, I do not think so. Commit 295e9178d3d74f151d87d56bed245f3931bdec2a does not have it, only the description of this PR (where a signed-off-by is not needed).

**Comment by yati1998** [comment]
> > > As mentioned on an other PR, I would really like to see your full name in the `Signed-off-by` instead of your GitHub username.
> > 
> > 
> > yes, I have done that already, in the commit message.
> 
> Hmm, I do not think so. Commit [295e917](https://github.com/ceph/ceph-csi/commit/295e9178d3d74f151d87d56bed245f3931bdec2a) does not have it, only the description of this PR (where a signed-off-by is not needed).

probably my bad.. I will change it.

File: docs/snap-clone.md
Comment (yati1998) [reply docs/snap-clone.md:53]: I guess its better to have it as mentioned in the contents. Let me know if you still want me to remove it. I will modify the content too.

File: docs/snap-clone.md
Comment (Madhu-1) [inline docs/snap-clone.md:10]: keep the table content same for both cephfs and rbd, currently it looks different

Comment (yati1998) [reply docs/snap-clone.md:10]: okay.

**Comment by Madhu-1** [comment]
@yati1998 can you please rebase this PR?

File: docs/snap-clone.md
Comment (nixpanic) [inline docs/snap-clone.md:168]: this contains output, not only commands, so it should be marked `console`

File: docs/snap-clone.md
Comment (nixpanic) [inline docs/snap-clone.md:49]: **snapshot** is owercase and **Volume** is uppercase. It would be nicer to have both **Snapshot** and **Volume**, here and below for RBD

**Review by nixpanic — DISMISSED** [review]
Looks good to me. I have commented on some nits that you could improve. In general, I also prefer to see Kubernetes objects like SnapshotClass with capitals in texts. This has not been done consistently from the beginning, so it is not required.

Unfortunately there have been some changes to the document, and this PR can not get merged without manual rebasing.

**Comment by yati1998** [comment]
@Madhu-1  and @nixpanic sure will make the requested changes and resolve the conflict too soon. Didn't resolve the conflict as I was waiting for the final review.

File: docs/snap-clone.md
Comment (yati1998) [reply docs/snap-clone.md]: done.

File: docs/snap-clone.md
Comment (yati1998) [reply docs/snap-clone.md]: done

**Review by nixpanic — APPROVED** [review]
Thanks!

**Pull request merged**
